File: backend/src/main/kotlin/com/example/domain/PokedexData.kt
package com.example.domain

data class PokedexData(
    val id: Long,
    val nameEn: String,
    val height: Float,
    val weight: Float,
    val types: List<Type>,
    val abilities: List<Ability>
) {
    companion object {
        fun of(pokemon: Pokemon, types: List<Type>, abilities: List<Ability>): PokedexData {
            return PokedexData(
                pokemon.id,
                pokemon.nameEn,
                pokemon.height,
                pokemon.weight,
                types,
                abilities
            )
        }
    }
}

File: frontend/models/pokedex-data.ts
import type Type from "../models/type";
import type Ability from "../models/ability";

interface PokedexData {
  id: string;
  nameEn: string;
  height: number;
  weight: number;
  types: Type[];
  abilities: Ability[];
}

export default PokedexData;
